packages/editor/src/lib/editor/managers/LicenseManager.ts

```diff
@@ -26,38 +26,57 @@ interface ValidLicenseKeyResult {
 }
 
 export class LicenseManager {
-	private publicKey = '3UylteUjvvOL4nKfN8KfjnTbSm6ayj23QihX9TsWPIM='
+	private publicKey: string
 	private isTest: boolean
-	constructor() {
+	constructor(testPublicKey?: string) {
 		this.isTest = typeof process !== 'undefined' && process.env.NODE_ENV === 'test'
+		this.publicKey = testPublicKey || '3UylteUjvvOL4nKfN8KfjnTbSm6ayj23QihX9TsWPIM='
 	}
-	private extractLicense(licenseKey: string): LicenseInfo {
-		const base64License = util.decodeBase64(licenseKey)
+	private async extractLicenseKey(licenseKey: string): Promise<LicenseInfo> {
+		const [data, signature] = licenseKey.split('.')
+		const [prefix, encodedData] = data.split('/')
 
-		const decoded = nacl.sign.open(base64License, util.decodeBase64(this.publicKey))
+		if (prefix !== 'tldraw') {
+			throw new Error(`Unsupported prefix '${prefix}'`)
+		}
+
+		const publicCryptoKey = await importPublicKey(this.publicKey)
 
-		if (!decoded) {
-			throw new Error('Invalid license key')
+		try {
+			await crypto.subtle.verify(
+				{
+					name: 'ECDSA',
+					hash: { name: 'SHA-384' },
+				},
+				publicCryptoKey,
+				new Uint8Array(str2ab(signature)),
+				new Uint8Array(str2ab(encodedData))
+			)
+		} catch (e) {
+			console.error(e)
+			throw new Error('Invalid signature')
 		}
-		const licenseInfo = JSON.parse(util.encodeUTF8(decoded))
-		return licenseInfoValidator.validate(licenseInfo)
+
+		const decodedData = JSON.parse(atob(encodedData))
```

Comment:
might need to put this in a try/catch too, in case the JSON parsing fails, quite possible